samples/sensor/dht_polling/src/main.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023 Ian Morris
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/kernel.h>
+
+#define DHT_ALIAS(i) DT_ALIAS(_CONCAT(dht, i))
+#define DHT_DEVICE(i, _)                                                                 \
+	IF_ENABLED(DT_NODE_EXISTS(DHT_ALIAS(i)), (DEVICE_DT_GET(DHT_ALIAS(i)),))
+
+/* Support up to 10 temperature/humidity sensors */
+static const struct device *const sensors[] = {LISTIFY(10, DHT_DEVICE, ())};
+
+int main(void)
+{
+	int rc;
+
+	for (size_t i = 0; i < ARRAY_SIZE(sensors); i++) {
+		if (!device_is_ready(sensors[i])) {
+			printk("sensor: device %s not ready.\n", sensors[i]->name);
+			return 0;
+		}
+	}
+
+	while (1) {
+		for (size_t i = 0; i < ARRAY_SIZE(sensors); i++) {
+			struct device *dev = (struct device *)sensors[i];
+
+			rc = sensor_sample_fetch(dev);
+			if (rc < 0) {
+				printk("%s: sensor_sample_fetch() failed: %d\n", dev->name, rc);
+				return rc;
+			}
+
+			struct sensor_value temp;
+			struct sensor_value hum;
+
+			rc = sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &temp);
+			if (rc == 0) {
+				rc = sensor_channel_get(dev, SENSOR_CHAN_HUMIDITY, &hum);
+			}
+			if (rc != 0) {
+				printf("get failed: %d\n", rc);
+				break;
+			}
+
+			printk("%16s: temp is %d.%d oC humidity is %d.%d %%RH\n", dev->name,
```

Comment:
```suggestion
			printk("%16s: temp is %d.%d °C humidity is %d.%d %%RH\n", dev->name,
```

Also, maybe the format should be `%d.%02d`, having so many digits feels odd.

Reply:
I changed ```oC``` to ```°C```, however when I tried the ```%d.%02d``` modification I found it did not do what I thought it would... After some reading it seems that (unless I have missed something) the width format specifier defines the minimum number of characters to be printed but it does not truncate the value if it is larger. Therefore there was no change in the output (in most cases) and so I left formatting as it was...

Reply:
Ah! of course... my bad. So also need to divide val2 by 10000, right?

https://github.com/zephyrproject-rtos/zephyr/blob/191e47bc8633a41ffd88ba64d740ea085bd68c0e/samples/shields/npm1300_ek/src/main.c#L96